waspc/cli/exe/Main.hs

```diff
@@ -122,6 +123,37 @@ dbCli args = case args of
   ["studio"] -> runDbCommand studio
   _ -> printDbUsage
 
+startCli :: [String] -> IO ()
+startCli args = case args of
+  [] -> runCommand $ start (StartOptions Nothing Nothing)
+  ["--web-client-port", clientPort, "--api-server-port", apiPort] -> runCommand $ start (StartOptions (Just clientPort) (Just apiPort))
+  ["--web-client-port", clientPort] -> runCommand $ start (StartOptions (Just clientPort) Nothing)
+  ["--api-server-port", apiPort] -> runCommand $ start (StartOptions Nothing (Just apiPort))
```

Comment:
What is tricky here is that you presume order of args. So right now, they can't put api-server-port first, and then web-client-port.

One way to go about this is to just search for them individually in this list.

So you might want to do smth like this:
```hs
let argPairs = zip args (drop 1 args)
let maybeClientPort = snd <$> find (\("--web-client-port", _) -> True) argPairs
let maybeServerPort = snd <$> find (\("--api-server-port", _) -> True) argPairs
case (maybeClientPort, maybeServerPort) of
  (Nothing, Nothing) -> printStartUsageWithArgs args
  _ -> runCommand $ start (StartOptions maybeClientPort maybeServerPort)
```

This is a bit rough code but I think it should give you an idea!

Reply:
Clever! :sunglasses: I've incorporated this change.  Thanks for the feedback.